gen/gen_table.go

```diff
@@ -191,7 +195,7 @@ func (p *pgClientImpl) list{{ .GoName }}(
 			return nil, p.client.errorConverter(&unstable.NotFoundError{
 				Msg: "Get{{ .GoName }}: record not found",
 			})
-		} else {
+		} else if !opt.SucceedOnPartialResults || len(ret) == 0 {
```

Comment:
I suppose this is really where we would want to drop the `|| len(ret) == 0` check.